src/main/java/de/tum/cit/aet/artemis/lecture/web/LectureUnitResource.java

```diff
@@ -200,11 +200,26 @@ public ResponseEntity<LectureUnitForLearningPathNodeDetailsDTO> getLectureUnitFo
      */
     @GetMapping("lecture-units/{lectureUnitId}")
     @EnforceAtLeastStudent
-    public ResponseEntity<LectureUnit> getLectureUnitById(@PathVariable @Valid Long lectureUnitId) {
+    public ResponseEntity<LectureUnit> getLectureUnitById(@PathVariable @Valid long lectureUnitId) {
         log.debug("REST request to get lecture unit with id: {}", lectureUnitId);
         var lectureUnit = lectureUnitRepository.findByIdWithCompletedUsersElseThrow(lectureUnitId);
         authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, lectureUnit.getLecture().getCourse(), null);
         lectureUnit.setCompleted(lectureUnit.isCompletedFor(userRepository.getUser()));
         return ResponseEntity.ok(lectureUnit);
     }
+
+    /**
+     * POST lectures/{lectureId}/lecture-units/{lectureUnitId}/ingest
+     * This endpooint is for starting the ingestion of one lecture unit.
+     *
+     * @param lectureUnitId The Id of the Lecture Unit that is <going to be ingested
+     * @return the ResponseEntity with status 200 (OK) and a message success or null if the operation failed
+     */
+    @PostMapping("lectures/{lectureId}/lecture-units/{lectureUnitId}/ingest")
+    @EnforceAtLeastInstructor
```

Comment:
The authorization check is not complete. This only checks that the user has the role instructor, but does not check if the user is instructor in the respective course. Please add additional checks in beginning of the method.

Reply:
You resolved the conversation, but my comment is still valid 🙈
I cannot see an authorization check based on the lectureId.
Also, you would need to double check that the lectureUnitId and the lectureId belong together